pkg/env/integersetting_test.go

```diff
@@ -0,0 +1,135 @@
+package env
+
+import (
+	"math"
+	"math/big"
+	"os"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIntegerSetting(t *testing.T) {
+	maxInt64PlusOne := big.NewInt(math.MaxInt64)
+	maxInt64PlusOne.Add(maxInt64PlusOne, big.NewInt(1))
+
+	cases := map[string]struct {
+		value        string
+		defaultValue int
+		minOpt       func() int
+		maxOpt       func() int
+		wantPanic    bool
+		wantValue    int
+	}{
+		"shall pass validation with no options": {
+			value:        "1",
+			defaultValue: 5,
+			wantPanic:    false,
+			wantValue:    1,
+		},
+		"shall pass validation with minimum": {
+			value:        "1",
+			defaultValue: 5,
+			minOpt:       func() int { return 0 },
+			wantPanic:    false,
+			wantValue:    1,
+		},
+		"shall fail validation with minimum": {
+			value:        "1",
+			defaultValue: 5,
+			minOpt:       func() int { return 5 },
+			wantPanic:    false,
+			wantValue:    5,
+		},
+		"shall panic with minimum": {
+			value:        "1",
+			defaultValue: 5,
+			minOpt:       func() int { return 10 },
+			wantPanic:    true,
+			wantValue:    1,
+		},
+		"shall pass validation with min and max": {
+			value:        "1",
+			defaultValue: 5,
+			minOpt:       func() int { return 0 },
+			maxOpt:       func() int { return 10 },
+			wantPanic:    false,
+			wantValue:    1,
+		},
+		"shall fail validation with min and max": {
+			value:        "11",
+			defaultValue: 5,
+			minOpt:       func() int { return 5 },
+			maxOpt:       func() int { return 10 },
+			wantPanic:    false,
+			wantValue:    5,
+		},
+		"shall panic with min and max": {
+			value:        "1",
+			defaultValue: 5,
+			minOpt:       func() int { return 10 },
+			maxOpt:       func() int { return 15 },
+			wantPanic:    true,
+			wantValue:    1,
+		},
+		"default value of min should have no influence if no WithMinimum is used explicitly": {
+			value:        "-5",
+			defaultValue: -1,
+			wantPanic:    false,
+			wantValue:    -5,
+		},
+		"options used incorrectly should panic": {
+			value:        "1",
+			defaultValue: 5,
+			minOpt:       func() int { return 10 },
+			maxOpt:       func() int { return 0 },
+			wantPanic:    true,
+			wantValue:    1,
+		},
+		"crossing the max/min value of int should return default value": {
+			value:        maxInt64PlusOne.Text(10),
```

Comment:
I think using `big.Int` is an overengineering. The previous string version for this test case was clear and good enough.

But if you prefer `big.Int`, why not just:
```suggestion
			value:        new(big.Int).Add(big.NewInt(math.MaxInt64), big.NewInt(1)).Text(10), // math.MaxInt + 1
```

Reply:
Here [ffd7b2f](https://github.com/stackrox/stackrox/pull/15799/commits/ffd7b2f4842f2635b1beffa20e3f425c62163401)